Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py

```diff
@@ -106,12 +110,31 @@ async def get_recent_trades_history(self, exchange, symbol):
     async def get_ohlcv_history(self, exchange, symbol, time_frame):
         # use time_frame_sec to add time to save the candle closing time
         time_frame_sec = commons_enums.TimeFramesMinutes[time_frame] * commons_constants.MINUTE_TO_SECONDS
-        candles = await self.exchange.get_symbol_prices(symbol, time_frame)
+
+        if self.start_timestamp is not None or self.end_timestamp is not None:
+            since = self.start_timestamp
+            if self.start_timestamp is None or self.start_timestamp < await self.get_first_candle_timestamp(symbol, time_frame):
```

Comment:
I think we should avoid calling `await self.exchange.get_symbol_prices` twice when possible, we could call it once and store it in `candle` and then figure out the first candle timestamp using it

Reply:
I made the `get_first_candle_timestamp` function to be able to use it from the web interface to have a minimum date for the datepicker of the start date. Do you want i remove it ?